homeassistant/components/conversation/default_agent.py

```diff
@@ -677,133 +655,108 @@ def _get_or_load_intents(
+                            intent_data_list = intent_dict.get("data", [])
+                            for intent_data in intent_data_list:
+                                sentence_metadata = intent_data.get("metadata", {})
+                                sentence_metadata[METADATA_CUSTOM_SENTENCE] = True
+                                sentence_metadata[METADATA_CUSTOM_FILE] = str(
+                                    custom_sentences_path.relative_to(
+                                        custom_sentences_dir.parent
                                     )
-                                    intent_data["metadata"] = sentence_metadata
-
-                            merge_dict(intents_dict, custom_sentences_yaml)
-                        else:
-                            _LOGGER.warning(
-                                "Custom sentences file does not match expected format path=%s",
-                                custom_sentences_file.name,
-                            )
-
-                    # Will need to recreate graph
-                    intents_changed = True
-                    _LOGGER.debug(
-                        "Loaded custom sentences language=%s (%s), path=%s",
-                        language,
-                        language_variant,
-                        custom_sentences_path,
-                    )
-
-            # Load sentences from HA config for default language only
-            if self._config_intents and (
-                self.hass.config.language in (language, language_variant)
-            ):
-                hass_config_path = self.hass.config.path()
-                merge_dict(
-                    intents_dict,
-                    {
-                        "intents": {
-                            intent_name: {
-                                "data": [
-                                    {
-                                        "sentences": sentences,
-                                        "metadata": {
-                                            METADATA_CUSTOM_SENTENCE: True,
-                                            METADATA_CUSTOM_FILE: hass_config_path,
-                                        },
-                                    }
-                                ]
-                            }
-                            for intent_name, sentences in self._config_intents.items()
-                        }
-                    },
-                )
-                intents_changed = True
+                                )
+                                intent_data["metadata"] = sentence_metadata
+
+                        merge_dict(intents_dict, custom_sentences_yaml)
+                    else:
+                        _LOGGER.warning(
+                            "Custom sentences file does not match expected format path=%s",
+                            custom_sentences_file.name,
+                        )
```

Comment:
Missing test coverage here

Reply:
This is existing lack of coverage. This change just changed the indentation of this block. We should address this in a follow-up PR.